prow/tide/tide_test.go

```diff
@@ -1964,3 +1964,28 @@ func TestPresubmitsByPull(t *testing.T) {
 		}
 	}
 }
+
+func TestApproverTags(t *testing.T) {
+	pr := PullRequest{
+		Reviews: struct {
+			Nodes []Review
+		}{
+			Nodes: []Review{
+				{Author: Author{Login: "dan", User: User{Email: "d@example.com"}}},
+				{Author: Author{Login: "carol", User: User{Name: "Carol", Email: "c@example.com"}}},
+				{Author: Author{Login: "bob", User: User{Name: "Bob"}}},
+				{Author: Author{Login: "alice"}},
+			},
+		},
+	}
+	fmt.Printf("%v\n", pr)
+	actual := pr.approverTags("Reviewed-by")
+	expected := `Reviewed-by: @alice
+Reviewed-by: @dan
+Reviewed-by: Bob
```

Comment:
Shouldn't we prefer `@bob` over Bob? Names are not unique like github logins

Reply:
> Shouldn't we prefer `@bob` over Bob?

Names are probably unique-enough within the context of the project.  And I expect a fair number of users will have both a name and email configured in GitHub, in which case we get the `name <email>` that Git uses for commits, and Git and Linux use for tags.  I'm fine falling back to `@login` instead of `name` in the no-email case, although I have a slight preference for names there.  Or maybe `name @login`?

Reply:
Putting GitHub usernames into commit messages is an anti-pattern and there is a PR in flight right now to disallow it -- it ends up spamming the mentioned users whenever a commit with that name is merged into any fork, ever.

Reply:
> Putting GitHub usernames into commit messages is an anti-pattern...

How about `GitHub $LOGIN` (no `@`) instead?

Reply:
> How about `GitHub $LOGIN` (no `@`) instead?

Done with a182510 -> 6e62cf4, which also rebases onto the current master (no conflicts) and uses `$NAME (GitHub $LOGIN)` when the name is non-empty (regardless of email presence) to address [the uniqueness concerns][1].  Check out the new unit tests for examples of each case.

[1]: https://github.com/kubernetes/test-infra/pull/9400#discussion_r218106548